### ISSUE TITLE
chore: Use npm 5.x for all Node.js Travis CI jobs

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#2734]
Historically we've kept the installed version of npm to be that of what is shipped with the Node.js version, this is partially incorrect and has been pointed out by npm that we should use the latest release of npm, the latest release of npm is `5.2.0`, npm does not maintain any LTS releases. npm will push security releases to previous 4.x and 3.x branches as required but no bug fixes.

More here: http://blog.npmjs.org/post/162451604515/the-npm-clis-long-term-support-lts-policy 

Once the v8 branch is merged I propose we switch to using npm 5.x for all Node.js versions.

I'm making a the @stylelint/core team recommendation that a prerequisite to use the latest npm 5.x release by all team members. (Basically if you are not using the latest npm `5.2.0` you are using out of date software)

I've created two pull requests demonstrating the changes I'll make after v8 is merged:
• https://github.com/stylelint/stylelint-config-standard/pull/84
• https://github.com/stylelint/stylelint-config-recommended/pull/3

**Comment by jeddy3** [comment]
SGTM

**Comment by evilebottnawi** [comment]
SGTM

**Comment by jeddy3** [comment]
@ntwb Do you have a moment to do this in this repo, or shall I?

**Comment by ntwb** [comment]
Thanks @jeddy3 

Rather than opening another issue, a couple of greenkeeper tasks are required (I think)

1. Via https://github.com/stylelint/stylelint/pull/2743#discussion_r127471267
> I think this will stop Greenkeeper from testing new versions of dependencies.
> Something like - `/^greenkeeper.*$/` might be needed to fix it.

2. Add https://github.com/greenkeeperio/greenkeeper-lockfile

I need to research these changes a little, have only quickly glanced at this issues so not fully understood here at my end yet :)

**Comment by ntwb** [comment]
I didn't notice that PR hadn't been merged, I've added both the above changes in https://github.com/stylelint/stylelint/pull/2743/commits/916ae50e068bc69b017911823eb23589b141588b
